### PR TITLE
Limit media files list + add get camera mode

### PR DESCRIPTION
**Pull request by nhanders**
@MikeWoots I snuck in get camera mode here too.

### Testing

✅  no limit to get file list
✅  limit get file list
✅  get camera mode

### CODE REVIEW

File: android/src/main/java/com/aerobotics/DjiMobile/DJIMedia.java [hunk 1]
```diff
@@ -111,8 +118,19 @@ public int compare(MediaFile lhs, MediaFile rhs) {
               }
             });
 
+            // Limit number of results
+            int resultsSize;
+            if (fileResultsLimit == null) {
+              resultsSize = mediaFileList.size();
+            } else if (fileResultsLimit.intValue() > mediaFileList.size()) {
```
Comment (MikeWoots) [inline android/src/main/java/com/aerobotics/DjiMobile/DJIMedia.java:125]: Is it possible to do this?
`if (fileResultsLimit == null || fileResultsLimit.intValue() > mediaFileList.size()) {
   resultsSize = mediaFileList.size();
}`

File: index.js [hunk 2]
```diff
@@ -173,8 +173,13 @@ const DJIMobileWrapper = {
   /**
    * ANDROID ONLY
    */
-  getFileList: async () => {
-    return await DJIMobile.getFileList();
+   getMediaFileList: async (numberOfResults: number = null) => {
+    console.log(numberOfResults);
```
Comment (MikeWoots) [inline index.js]: Can we remove this logging?

File: android/src/main/java/com/aerobotics/DjiMobile/DJIMobile.java [hunk 3]
```diff
@@ -149,17 +149,32 @@ public void onDatabaseDownloadProgress(long l, long l1) {
...
+      m.getMediaFileList(promise, product);
+    }
+  }
+
+  @ReactMethod
+  public void getLimitedMediaFileList(Integer numberOfResults, final Promise promise) {
+    DJIMedia m = new DJIMedia(reactContext);
+    if (product == null) {
+        product = DJISDKManager.getInstance().getProduct();
+        if (product == null) {
```
Comment (MikeWoots) [inline android/src/main/java/com/aerobotics/DjiMobile/DJIMobile.java]: I know you where just using the same logic as the other method but I think this second null check is redundant. As in if the first check fails the second is also going to fail. The product variable gets set when the drone connects so if it's null then trying instantiate it using `DJISDKManager.getInstance().getProduct()` is ultimately going to return null as well

File: android/src/main/java/com/aerobotics/DjiMobile/DJIMedia.java [hunk 4]
```diff
@@ -111,12 +118,23 @@ public int compare(MediaFile lhs, MediaFile rhs) {
...
             WritableArray params = Arguments.createArray();
-            for (MediaFile m : mediaFileList){
+            for (int fileIndex = 0; fileIndex < resultsSize; fileIndex++) {
+              MediaFile m  = mediaFileList.get(fileIndex);
               WritableMap file = Arguments.createMap();
               file.putString("fileName", m.getFileName());
-              file.putDouble("timeCreatedAt", m.getTimeCreated());
-              file.putDouble("fileSize", m.getFileSize());
+              file.putString("dateCreated", m.getDateCreated());
+              file.putDouble("fileSizeInBytes", m.getFileSize());
```
Comment (MikeWoots) [inline android/src/main/java/com/aerobotics/DjiMobile/DJIMedia.java:137]: @NickHAnderson019 is this not potentially going to break something on the JS or are you going to update that as well? (I suppose we aren't actually using this for anything yet)

File: index.js [hunk 5]
```diff
@@ -58,6 +58,12 @@ export type FlightLogListenerEvent = {
   type: string,
 }
 
+export type mediaFileData = {
```
Comment (MikeWoots) [inline index.js]: Can you capitalise this? MediaFileData

**Review by MikeWoots — APPROVED** [review]

File: index.js [hunk 6]
```diff
@@ -58,7 +58,7 @@ export type FlightLogListenerEvent = {
   type: string,
 }
 
-export type mediaFileData = {
+export type MediaFileData = {
```
Comment (nhanders) [inline index.js:61]: Thanks @MikeWoots 😄

**Pull request merged**